Assets/Scripts/Interactables/Dialogue/SpeakerUI.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine.UI;
+using UnityEngine;
+using TMPro;
+
+public class SpeakerUI : MonoBehaviour
+{
+    public TMP_Text dialogue;
+    public GameObject textBoxContainer;
+    
+    private Vector3 textPosition;
+
+    void Start()
+    {
+        this.Hide();
+    }
+
+    public string Dialogue
+    {
+        set { dialogue.text = value;}
+    }

```

Comment:
This isn't a function, so can we move this like at the top with the other variables?